File: backend/user-service/src/main/java/com/healthpartner/users/dto/UserProfileDto.java
package com.healthpartner.users.dto;

import com.healthpartner.users.model.BMICategory;
import com.healthpartner.users.model.Gender;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserProfileDto {

    private int userId;
    private String userName;
    private String surName;
    private String emailId;
    private String address;
    private String mobileNo;
    private int age;
    private Gender gender;
    private double height;
    private double weight;
    private double bmiIndex;
    private BMICategory bmiCategory;

    private String profilePicUrl;
    private Date createdAt;
    private Date updatedAt;

}

File: frontend/src/app/models/user-profile.dto.ts
import { Gender, BMICategory } from './user.dto';

export interface UserProfileDto {
  userId: number;
  userName: string;
  surName: string;
  emailId: string;
  address: string | null;
  mobileNo: string | null;
  age: number;
  gender: Gender;
  height: number;
  weight: number;
  bmiIndex: number;
  bmiCategory: BMICategory;
  profilePicUrl: string | null;
  createdAt: Date;
  updatedAt: Date;
} 